plenum/server/catchup/cons_proof_service.py

```diff
@@ -99,6 +99,14 @@ def process_consistency_proof(self, proof: ConsistencyProof, frm: str):
     def _finish(self, cons_proof: Optional[ConsistencyProof] = None):
         logger.info("{} finished with consistency proof {}".format(self, cons_proof))
 
+        till = CatchupTill(start_size=cons_proof.seqNoStart,
+                           final_size=cons_proof.seqNoEnd,
+                           final_hash=cons_proof.newMerkleRoot) if cons_proof else None
+
+        nodes_ledger_sizes = {frm: proof.seqNoEnd
```

Comment:
Why don't we just filter nodes according to the following simple rule:
`get all nodes we have ConsistencyProof from with a `seqNoEnd` not less than `seqNoEnd` we choice for catchup`?

Reply:
If we do such filtering then most probably we'll filter out all nodes that have less transactions than majority of pool, severely limiting number of nodes participating in catch up. With strategy implemented we actually give chance to all nodes that have new transactions to participate.